### DISCUSSION TITLE
Is the S2 library based on the WGS-84 coordinate system?

### DISCUSSION THREAD

**Discussion by LemonLiTree** [google/s2geometry#311] [Q&A] [answered]
I looked at the official documents, but I didn't find any instructions about WGS-84. I want to confirm it.

**Comment by jmr** [comment]
No, everything is based on a sphere.  `s2earth.h` has a radius, but I'm not sure how it compares to the equivalent radius of the WGS 84 spheroid.

**Comment by smcallis** [comment]
s2earth radius is the mean radius which averages the difference between
polar and equatorial radius.  It's 6,371.0088 km which is the same as the
WGS-84 mean radius.

**Reply by LemonLiTree** [reply]
@smcallis Can I think that the geographical calculation results based on the S2 library are in line with the WGS 84 coordinate system？

**Reply by smcallis** [reply, marked as answer]
No in general no because there's no ellipsoidal correction, so distances and areas won't be the same.  The error is small but noticeable.

**Reply by LemonLiTree** [reply]
last question：Is the geography_functions in bigquery implemented based on the S2 library?
https://cloud.google.com/bigquery/docs/reference/standard-sql/geography_functions

**Comment by smcallis** [comment]
Those are implemented based on a library called STLib which is built on top
of S2 but isn't open sourced (yet).

**Reply by LemonLiTree** [reply]
thks
